src/main/java/strategies/impl/AllStrategyImpl.java

```diff
@@ -18,8 +19,9 @@
                 .forEach(wordIndexes::retainAll);
 
         return linesIndexesByWord.keySet().stream()
-                .filter(word -> linesIndexesByWord.getOrDefault(word, new ArrayList<>()).containsAll(wordIndexes))
-                .flatMap(word -> linesIndexesByWord.getOrDefault(word, new ArrayList<>()).stream())
+                .map(matchedWord -> linesIndexesByWord.getOrDefault(matchedWord, new ArrayList<>()))
```

Comment:
`keySet()` contains key-value pairs, so you don't need to use `linesIndexesByWord.getOrDefault(matchedWord, new ArrayList<>())` for getting lines indexes.